### PR TITLE
Move most of the root package into a subcrate.

### PR DESCRIPTION
**Pull request by thegedge**
This improves incremental build time when working on what was previously the root package. For example, previously all plugins would be rebuilt with a change to `src/commands/classified/external.rs`, but now only `nu-cli` will have to be rebuilt (and anything that depends on it).

### CODE REVIEW

**Comment by thegedge** [comment]
CI failures make no sense. `sample.db` is there 🤔

**Comment by jonathandturner** [comment]
@thegedge - looks like you're deleting it in the PR?

Screenshot from 2020-03-04 05-06-57

**Comment by thegedge** [comment]
Weird. I have the file locally, but for some reason it isn't being committed. I bet I have it in my global gitignore...

File: crates/nu-test-support/src/playground.rs [hunk 1]
```diff
@@ -59,9 +59,7 @@ impl Playground {
         let fixtures = fixtures
             .parent()
             .expect("Couldn't find the fixtures directory")
-            .parent()
-            .expect("Couldn't find the fixtures directory")
-            .join("tests/fixtures");
+            .join("nu-cli/tests/fixtures");
```
Comment (andrasio) [inline crates/nu-test-support/src/playground.rs]: try ` .join("crates/nu-cli/tests/fixtures");` instead.

Comment (andrasio) [reply crates/nu-test-support/src/playground.rs]: Seems it's working without this.

**Comment by thegedge** [comment]
Oh yeah, `cargo install` is still good:

```
$ cargo install --features=stable --path .
...
...
...
    Replaced package `nu v0.10.1 (/Users/gedge/src/github.com/nushell/nushell)` with `nu v0.10.1 (/Users/gedge/src/github.com/nushell/nushell)` (executables `nu`, `nu_plugin_core_inc`, `nu_plugin_core_ps`, `nu_plugin_core_str`, `nu_plugin_core_sys`, `nu_plugin_core_textview`, `nu_plugin_stable_average`, `nu_plugin_stable_binaryview`, `nu_plugin_stable_fetch`, `nu_plugin_stable_match`, `nu_plugin_stable_post`, `nu_plugin_stable_shuffle`, `nu_plugin_stable_sum`, `nu_plugin_stable_tree`)
```

**Pull request merged**